### ISSUE TITLE
Request merge x and y windows

### ISSUE THREAD

**Issue by rayman89** [jonathanKingston/merge-windows#1]
Could you add the option to merge some windows and not all?

**Comment by jonathanKingston** [comment]
I'm not sure how this could be listed, perhaps listing the current window titles?

This would also make the current menu item be nested under a child menu. There is only two levels in the context menus so the menu would have to look something like:

- Take a screenshot
- Merge windows
  - All windows
  - ---
  - With window (50) Facebook window
  - With window (25) Gmail

**Comment by rayman89** [comment]
Yes, windows titles is what I've seen other extension use to refer to other windows. What you suggested sounds pretty nice. Thank you!

**Comment by AdamPS** [comment]
Similar request please: I would like to see a keyboard shortcut (I use Ctrl+Shift+G) that merges the current window with the other most recently active window.  This was popular in TabMixPlus before Quantum.  I would donate for this one, thanks.

**Comment by minionsweb** [comment]
ditto!, often have hundreds of tabs in many windows (18 windows currently open with 2500 tabs between them) while researching subjects.
Being able to specify 2 or more windows to merge togetherwould go a great way towards organizing and bookmarking groups of like information.

I will look into tabmixplus using waterfox for a temporary solution (also great since I can still use session manager in waterfox)

**Comment by AdamPS** [comment]
Seriously, I am willing to pay for this - send me a message and we can discuss how much.  I feel it would be better in this add-on, but if the maintainer here isn't interested then I will offer it more widely for someone to create a new one.

I would like a key that directly merges the current and prev windows. I'm not a Mozilla coder, but it looks like the "previous focus window" could be tracked something like this.  

```
var curFocusWindowId;
var prevFocusWindowId;

browser.windows.onFocusChanged.addListener((windowId) => {
  prevFocusWindowId = curFocusWindowId;
  curFocusWindowId = windowId;
});
```

**Comment by AdamPS** [comment]
I posted a [bounty](https://www.bountysource.com/issues/46535415-request-merge-x-and-y-windows).  To claim the bounty, the solution should include a key to merge the 2 most recent windows.

**Comment by Zegnat** [comment]
I have started a Draft PR in #8 that addresses both the more specific merging menu and the keyboard shortcuts points mentioned in this ticket. Still a draft as I am not completely sold on my implementation yet, but I would love to get some feedback and testers on it!

**Comment by AdamPS** [comment]
@Zegnat many thanks for taking on this issue.  I am a developer but have never tried Browser add-on development so I don't have any knowledge of building them.  Please could you post the compiled add-on file or send it to me by private message?

**Comment by Zegnat** [comment]
I do not have my AMO credentials on this machine so I can’t get the add-on compiled (which requires signing by Mozilla). Should be able to do so tomorrow.

Alternatively, to run an in-development add-on yourself:

1. Clone the code to your machine (from [my repo, get the correct branch](https://github.com/Zegnat/browserext-merge-windows/tree/pick-a-window)),
2. in Firefox go to `about:debugging` (you may need the Developer or Nightly edition) and click the blue text. (“This Nightly” in “Use This Nightly to debug extensions <i>[…]</i>” in the Nightly edition.)
3. Click the “Load Temporary Add-on…” button and select the `manifest.json` in the directory you cloned in step 1.
4. You are now good to go.

Follow the instructions in #8 if you need help configuring keyboard shortcuts.

**Comment by Zegnat** [comment]
I have abandoned my previously mentioned PR. Instead I have [rebranded my fork](https://github.com/Zegnat/browserext-window-merger) and am providing [my own up-to-date add-on](https://addons.mozilla.org/firefox/addon/window-merger/) that supports the behaviour discussed in this issue.

Open-source software is never dependent on a single party 😄

**Comment by AdamPS** [comment]
Thanks @Zegnat I confirm that the new fork fixes this bug.
